java/code/src/com/redhat/rhn/domain/channel/Channel.java

```diff
@@ -234,14 +257,27 @@ public void cloneModulesFrom(Channel from) {
     }
 
     /**
+     * Gets the synced module metadata files belonging to the channel
+     * <p>
+     * See {@link Channel#getLatestModules()} to get the latest metadata file currently in use.
+     *
      * @return Returns the Modules.
      */
-    public Modules getModules() {
+    public Set<Modules> getModules() {
         return modules;
     }
 
+    /**
+     * Gets the latest module metadata file in use
+     *
+     * @return the module metadata (modules.yaml) file
+     */
+    public Modules getLatestModules() {
+        return modules.stream().max(Comparator.comparing(Modules::getLastModified)).orElse(null);
```

Comment:
I let a comment in the database transformation. But now I understand why you need to keep the old last modified date when cloning channels. You need it here. (I left the other comment anyway, so you can check it).